boltstub/wiring.py

```diff
@@ -251,10 +352,50 @@ def remote_address(self):
         """
         return Address(self.__socket.getpeername())
 
+    @property
+    def _socket(self):
+        return self.__socket
+
 
 class WireError(OSError):
     """Raised when a connection error occurs."""
 
 
 class BrokenWireError(WireError):
     """Raised when a connection is broken by the network or remote peer."""
+
+
+def negotiate_socket(_socket):
+    def _try_to_negotiate_websocket(__socket, __buffer):
```

Comment:
About the function name:  
No need to mark the function as internal (starting with `_`). It's scoped to it's parent function so it's not accessible from the outside anyway.

About the parameter names:  
(PEP8)[https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles] is not explicit on this but it states
 * > `_single_leading_underscore`: weak "internal use" indicator. E.g. `from M import *` does not import objects whose names start with an underscore.

 So this only makes sense for module globals as well as class/object attributes/methods.

 * > `single_trailing_underscore_`: used by convention to avoid conflicts with Python keyword, e.g.

So I suggest to use a single trailing underscore to avoid shadowing, e.g., `socket_` instead of `_socket`.


This comment applies to other places as well (e.g. `negotiate_socket(_socket)` just above).

Reply:
Sure, I will fix up. It was my JS accent talking.